modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/InterceptingExecutionMediator.java

```diff
@@ -39,8 +42,8 @@ public InterceptingExecutionMediator(ExecutionMediator intercepted) {
   }
 
   @Override
-  public Object execute(OperationExecutor executor, ExecutionContextAdapter context) throws Throwable {
-    Object resultValue = intercepted.execute(executor, context);
+  public Mono<Object> execute(OperationExecutor executor, ExecutionContextAdapter context) throws Throwable {
+    Object resultValue = intercepted.execute(executor, context).block();
```

Comment:
Why block here.  Seems you are blocking and then using `just` to create Mono from value later?